### PR TITLE
Fix user can be born in the future (#327)

### PR DESCRIPTION
**Pull request by e-n-0**

### CODE REVIEW

**Review by Nhqml — APPROVED** [review]

**Comment by Elfikurr** [comment]
I believe that `clean()` should raise a `ValidationError` when encountering an error ?
Especially if there is only one error, else we should use `add_error()` instead of `self.error_class`, if I'm reading the docs (badum tss) correctly.

File: prologin/users/forms.py [hunk 1]
```diff
@@ -104,7 +105,14 @@ def clean(self):
         if not self.can_edit_profile:
             for field in self.readonly_during_contest:
                 self.cleaned_data.pop(field, None)
-        return super().clean()
+        
+        super().clean()
+
+        birthday = self.cleaned_data.get('birthday')
+        if birthday and birthday > datetime.today().date():
```
Comment (lportemo) [inline prologin/users/forms.py]: isn't there any warning about using datetime.today().date() instead of timezone.now().date() ?

**Review by lportemo — CHANGES_REQUESTED** [review]
Why can't you just use a validator on the model ?

File: prologin/users/forms.py (hunk 1 above)
Comment (e-n-0) [reply prologin/users/forms.py]: Using user's timezone is maybe better.

**Comment by Nhqml** [comment]
IMO, the use of validators as suggested by @lportemo is a cleaner way to achieve our goal. It will also be more consistent if we decide to implement more validations steps in the future.

I believe we should instead focus on #331. Feel free to reopen this if you disagree.

**Pull request closed without merging**